regions_www/m/server.R

```diff
@@ -234,9 +234,11 @@ shinyServer(function(input, output, session) {
     # Rm objects (by time last accessed) if the list size is more than 2 Gb
     if (format(object.size(loaded_data), units = "Gb") > 2) {
       idx_to_rm <- which(loaded_data_accessed == min(unlist(loaded_data_accessed)))
-      print(c("removing", dir_to_rm, format(object.size(loaded_data), units = "Gb")))
-      loaded_data[[idxes_to_rm]] <<- NULL
-      loaded_data_accessed[[idxes_to_rm]] <<- NULL
+      if (interactive()){
```

Comment:
What's the downside of printing results when not running interactively? I just checked this by adding

```
  if(interactive())  print("hello")
```

in the `shiny::runExample("01_hello")` example and it does indeed print. Happy with this, just curious as to why you've made printing conditional.

Reply:
Just I was nervous about over logging. When run via shiny server these print statements don't appear in the log and I don't think they'd be useful in the server logs.

Reply:
Understood, but do server logs contain outputs from `print()` commands? I tested it in interactive and non-interactive settings and cannot see the the print messages when calling Shiny from a non interactive environment. To test you can try this from R:

```r
# no message
shiny::runGitHub(repo = "rstudio/shiny", subdir = "inst/examples/01_hello")
# message from conditional and non conditional print
shiny::runGitHub(repo = "robinlovelace/shiny", subdir = "inst/examples/01_hello", ref = "patch-1")
```

And this from bash:

```r
 # prints message
Rscript -e 'print("hello")
# no message:
Rscript -e 'shiny::runGitHub(repo = "robinlovelace/shiny", subdir = "inst/examples/01_hello", ref = "patch-1")' 
```

Reply:
Thanks @Robinlovelace, good point I'm not sure exactly if shiny puts `print` statements into the log.  With your example I wouldn't expect to see the `print` statement in the console, I'd expect it in the log file.  Even shiny doesn't put `print` statements into the logs we don't need these in production.  Does that answer you question about why I made them conditional?  I added them for local debugging only so I didn't want the server to waste resources (albeit tiny) on running them.

Reply:
I vote remove the conditionality but cannot see a downside.